src/envoy/http/alpn/alpn_test.cc

```diff
@@ -31,55 +34,139 @@ namespace {
 
 class AlpnFilterTest : public testing::Test {
  public:
-  std::unique_ptr<AlpnFilter> makeDefaultFilter() {
-    auto default_config = std::make_shared<AlpnFilterConfig>();
-    auto filter = std::make_unique<AlpnFilter>(default_config);
-    filter->setDecoderFilterCallbacks(callbacks_);
-    return filter;
-  }
-
-  std::unique_ptr<AlpnFilter> makeAlpnOverrideFilter(
-      const std::vector<std::string> &alpn) {
+  std::unique_ptr<AlpnFilter> makeAlpnOverrideFilter(const AlpnOverride &alpn) {
     FilterConfig proto_config;
-    for (const auto &protocol : alpn) {
-      proto_config.add_alpn_override(protocol);
+
+    for (const auto &p : alpn) {
+      FilterConfig_AlpnOverride entry;
+      entry.set_upstream_protocol(getProtocol(p.first));
+      for (const auto &v : p.second) {
+        entry.add_alpn_override(v);
+      }
+      proto_config.mutable_alpn_override()->Add(std::move(entry));
     }
-    auto config = std::make_shared<AlpnFilterConfig>(proto_config);
+
+    auto config =
+        std::make_shared<AlpnFilterConfig>(proto_config, cluster_manager_);
     auto filter = std::make_unique<AlpnFilter>(config);
     filter->setDecoderFilterCallbacks(callbacks_);
     return filter;
   }
 
  protected:
+  FilterConfig::Protocol getProtocol(Http::Protocol protocol) {
```

Comment:
Could you elaborate? I don't really understand why you don't use Http::Protocol? It's 1:1 mapping and I see the conversion back and forth.

Reply:
I defined and used a set of protocols in the proto file.